app/controllers/payment-links/metadata/metadata-form.js

```diff
@@ -1,19 +1,42 @@
+const META_DATA_MAX_LENGTH = 50
+const MAX_NO_OF_METADATA_COLUMNS = 10
+
 const fields = {
   metadataKey: {
     id: 'metadata-column-header',
     name: 'metadata-column-header',
-    validation: [ {
-      validator: isNotEmpty,
-      message: 'Enter a column header'
-    } ]
+    validation: [
+      {
+        validator: isNotEmpty,
+        message: 'Enter a column header'
+      },
+      {
+        validator: isValidLength,
+        message: `Column header must be ${META_DATA_MAX_LENGTH} characters or fewer`
+      },
+      {
+        validator: isNotDuplicate,
+        message: 'Column header must not already exist'
+      },
+      {
+        validator: isNotExceedingMaxNoOfReportingColumns,
+        message: `Number of reporting columns must be ${MAX_NO_OF_METADATA_COLUMNS} or fewer`
+      }
+    ]
   },
   metadataValue: {
     id: 'metadata-cell-value',
     name: 'metadata-cell-value',
-    validation: [ {
-      validator: isNotEmpty,
-      message: 'Enter cell content'
-    } ]
+    validation: [
+      {
+        validator: isNotEmpty,
+        message: 'Enter cell content'
+      },
+      {
+        validator: isValidLength,
+        message: 'Cell content must be 50 characters or fewer'
```

Comment:
This message should use the variable that's used for valid length

Reply:
Good spot - done.